controllers/azureidentity_controller.go

```diff
@@ -162,7 +162,7 @@ func (r *AzureIdentityReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		// only delete bindings when the identity owner type is not found.
 		// we should not delete an identity when azureCluster is not found because it could have been created by AzureManagedControlPlane.
 		switch identityOwner.(type) {
-		case infrav1.AzureCluster:
+		case *infrav1.AzureCluster:
```

Comment:
This wouldn't hit this line unless it was a pointer.  Changing it to a pointer covers the line and ensure the object we get back from the assertion is in fact a client.Object

Reply:
Was this line not working before this change? Great catch if so!